ingress/controllers/nginx/configuration.md

```diff
@@ -0,0 +1,394 @@
+## Contents
+* [Customizing NGINX](#customizing-nginx)
+* [Custom NGINX configuration](#custom-nginx-configuration)
+* [Custom NGINX template](#custom-nginx-template)
+* [Annotations](#annotations)
+* [Custom NGINX upstream checks](#custom-nginx-upstream-checks)
+* [Authentication](#authentication)
+* [Rewrite](#rewrite)
+* [Rate limiting](#rate-limiting)
+* [Secure NGINX upstreams](#secure-nginx-upstreams)
```

Comment:
Or is there motivation for calling it "upstreams" that I'm missing (either in nginx or if you want to keep upstreams for some reason)?